src/operator/src/statement/ddl.rs

```diff
@@ -74,6 +78,47 @@ impl StatementExecutor {
             .await
     }
 
+    #[tracing::instrument(skip_all)]
+    pub async fn create_table_like(
+        &self,
+        stmt: CreateTableLike,
+        ctx: QueryContextRef,
+    ) -> Result<TableRef> {
+        let (catalog, schema, table) = table_idents_to_full_name(&stmt.target, &ctx)
+            .map_err(BoxedError::new)
+            .context(error::ExternalSnafu)?;
+        let table_ref = self
+            .catalog_manager
+            .table(&catalog, &schema, &table)
+            .await
+            .context(error::CatalogSnafu)?
+            .context(error::TableNotFoundSnafu { table_name: &table })?;
+        let partitions = self
+            .partition_manager
+            .find_table_partitions(table_ref.table_info().table_id())
+            .await
+            .context(error::FindTablePartitionRuleSnafu { table_name: table })?;
+
+        let quote_style = ctx.quote_style();
+        let mut create_stmt =
+            show_create_table::create_table_stmt(&table_ref.table_info(), quote_style)
+                .context(error::ParseQuerySnafu)?;
+        create_stmt.name = stmt.name;
+        create_stmt.if_not_exists = false;
```

Comment:
Looks like `create table like` can support `if_not_exists` too, just like

```sql
CREATE TABLE IF NOT EXISTS snapshot LIKE source_data;
```

Reply:
In fact, I thought so at first, but I checked MySQL's `Create Table ... Like` and it seems that it is not supported, so I chose to keep it unified. Of course, this is up to you to decide.

ref: https://dev.mysql.com/doc/refman/8.0/en/create-table-like.html

Reply:
Got it. Let's leave it currently.